### PR TITLE
BACKLOG-19987: Allow to customize accordions used by a picker

### PR DESCRIPTION
**Pull request by cedmail**
…

## JIRA

https://jira.jahia.org/browse/BACKLOG-19987

## Description

Allow to customize accordions used by a picker directly from the picker. Allow to pass accordionItemsProps directly from the picker config

https://docs.google.com/document/d/1WyI_EMJA9K6r6NYJzC7ZJiE-yCJBuMcITGrBW-vf5IE/edit#

### CODE REVIEW

File: src/javascript/SelectorTypes/Picker/PickerDialog/SelectionHandler.jsx [hunk 1]
```diff
@@ -87,7 +87,16 @@ export const SelectionHandler = ({initialSelectedItem, editorContext, pickerConf
...
-        const allAccordionItems = jcontentUtils.getAccordionItems(pickerConfig.key, accordionItemProps);
+        let allAccordionItems = jcontentUtils.getAccordionItems(pickerConfig.key, accordionItemProps);
+
+        if (allAccordionItems.length === 0 && pickerConfig.accordions !== undefined && pickerConfig.accordions.length > 0) {
+            allAccordionItems = [];
+            pickerConfig.accordions.forEach(value => {
+                const accordionItem = registry.get('accordionItem', value);
+                accordionItem.targets.push({id: pickerConfig.key, priority: 50});
+                allAccordionItems.push(jcontentUtils.getAccordionItem(accordionItem, accordionItemProps));
+            });
```
Comment (tdraier) [inline src/javascript/SelectorTypes/Picker/PickerDialog/SelectionHandler.jsx:98]: We should do that only once, maybe in a initCallback after all picker and accordions have been registered ?

Comment (cedmail) [reply src/javascript/SelectorTypes/Picker/PickerDialog/SelectionHandler.jsx:98]: good idea :)

File: src/javascript/SelectorTypes/Picker/PickerDialog/RightPanel/RightPanel.jsx [hunk 2]
```diff
@@ -51,10 +51,10 @@ const RightPanel = ({pickerConfig, accordionItemProps, onClose, onItemSelection}
                 </div>
             </header>
             <div className={clsx('flexFluid', 'flexCol_nowrap', css.body)}>
-                {mode !== '' && <ContentLayout pickerConfig={pickerConfig}/>}
+                {mode !== '' && <ContentLayout pickerConfig={pickerConfig} accordionItemProps={accordionItemProps}/>}
             </div>
 
-            <SelectionTable selection={selection} expanded={selectionExpanded} pickerConfig={pickerConfig}/>
+            <SelectionTable selection={selection} expanded={selectionExpanded} pickerConfig={pickerConfig} accordionItemProps={accordionItemProps}/>
```
Comment (tdraier) [inline src/javascript/SelectorTypes/Picker/PickerDialog/RightPanel/RightPanel.jsx:57]: That's probably not needed here ?

File: src/javascript/SelectorTypes/Picker/PickerDialog/SelectionHandler.jsx [hunk 3]
```diff
@@ -87,7 +87,7 @@ export const SelectionHandler = ({initialSelectedItem, editorContext, pickerConf
 
         const selectedNode = nodesInfo.data && nodesInfo.data.jcr.nodesByPath.length > 0 && nodesInfo.data.jcr.nodesByPath[0];
 
-        const allAccordionItems = jcontentUtils.getAccordionItems(pickerConfig.key, accordionItemProps);
+        let allAccordionItems = jcontentUtils.getAccordionItems(pickerConfig.key, accordionItemProps);
```
Comment (tdraier) [inline src/javascript/SelectorTypes/Picker/PickerDialog/SelectionHandler.jsx:90]: const ?

File: src/javascript/SelectorTypes/Picker/Picker2.jsx [hunk 4]
```diff
@@ -138,7 +138,7 @@ export const Picker2 = ({field, value, editorContext, inputContext, onChange, on
                     editorContext={editorContext}
                     pickerConfig={pickerConfig}
                     initialSelectedItem={fieldData && fieldData.map(f => f.path)}
-                    accordionItemProps={parsedOptions.accordionItem}
+                    accordionItemProps={parsedOptions.accordionItem === undefined ? pickerConfig.accordionItem : parsedOptions.accordionItem}
```
Comment (tdraier) [inline src/javascript/SelectorTypes/Picker/Picker2.jsx:141]: I'm just not sure if the behaviour would be ok if you have field override + accordion config in picker config ? I would expect override to just extend the normal config, not replace it

**Review by tdraier — APPROVED** [review]

**Pull request merged**